### ISSUE TITLE
[invader-bitmap] Bump map palette

### ISSUE THREAD

**Issue by hellux** [SnowyMouse/invader#205]
Hey, I am reading 8-bit bump maps from the H1X game files. From what I understand, the textures are palettized such that the 8-bit color is just an index to a color in a configurable look-up table. So I assume the game loads a palette and the textures to the GPU so the GPU can perform the lookup.

I'm not really sure where the game gets this palette from but I noticed that there is a palette in this repo (src/bitmap/p8/p8_palette) that seems to work well and is from the Xbox according to the commit message. So how was this palette obtained? Can it be found in the map files or in the game binary?

Thanks!

**Comment by SnowyMouse** [comment]
Hello! I got the palette from the MEK which you can get from here: https://github.com/Sigmmma/reclaimer/blob/master/reclaimer/bitmaps/p8_palette_halo

As for how to find this palette, you could look for it in tool.exe's code or guerilla.exe's code. In guerilla.exe, it's stored at 0x5B9A80.

**Comment by SnowyMouse** [comment]
Alternatively, you could try making Guerilla preview a bitmap with all 256 values in it and use a screenshot of it to generate a palette.

You can do this by making a Y8 bitmap with all colors from 000000 to FFFFFF (000000, 010101, 020202, ..., FFFFFF).

![image](https://user-images.githubusercontent.com/39858815/116331096-a4fd6380-a794-11eb-8c60-58ae8ef4fcb4.png)

Next, take a screenshot of the alpha channel and color channel in Guerilla and put it in an image editor. I'll use Krita for this. Save the transparency mask merged and you'll get an image like this:

![merged](https://user-images.githubusercontent.com/39858815/116331221-e3931e00-a794-11eb-9334-d98b99bfc1a4.png)

Decode the pixels and swap the channel order and you'll get the palette.

![image](https://user-images.githubusercontent.com/39858815/116331663-dfb3cb80-a795-11eb-9216-94ae3f81f10c.png)

**Comment by hellux** [comment]
I was able to find the palette at 0x5b9a80 in guerilla.exe. It seems to be ordered as BGRA instead of ARGB as in your file. I searched for this one in the Xbox binaries and managed to find exact matches at:
```
default.xbe: 0x22dc20 (01.10.12.2276, NTSC)
default.xbe: 0x227f78 (01.01.14.2342, PAL)
```
I was hoping to load the palette from the map file at runtime but I guess I'll just embed it into the binary like they did.

Thanks for the pointer!
